service/src/main/java/tds/config/services/ConfigService.java

```diff
@@ -16,22 +17,73 @@
      */
     Optional<ClientTestProperty> getClientTestProperty(String clientName, String assessmentId);
 
     /**
      * Get the {@link ClientSystemFlag} for the specified client and audit name.
      * <p>
-     *     The audit name is the name/description of the ClientSystemFlag.  Ideally the source of data for this method
-     *     will be an in-memory collection; the {@code ClientSystemFlag} values rarely change, making them a strong
-     *     candidate for caching.
+     *     The audit name is the name/description of the {@link ClientSystemFlag}.  Ideally the source of data for this
+     *     method will be an in-memory collection; the {@link ClientSystemFlag} values rarely change, making them a
+     *     strong candidate for caching.
      * </p>
      *
      * @param clientName  The name of the client (typically SBAC or SBAC_PT)
-     * @param auditObject The name of the desired {@code ClientSystemFlag}
-     * @return The {@code ClientSystemFlag} that matches the client and audit name.
-     * @throws NoSuchElementException if the {@code ClientSystemFlag} with the specified audit name does not exist
+     * @param auditObject The name of the desired {@link ClientSystemFlag}
+     * @return The {@link Optional<ClientSystemFlag>} that matches the client and audit name.
      */
     Optional<ClientSystemFlag> getClientSystemFlag(String clientName, String auditObject);
+
+    /**
+     * Get the TA Check-In time limit for the specified client.
+     * <p>
+     *     This is a special method to fetch the {@code tacheckintime} from the {@code session.timelimits} view.  Unlike
+     *     the other timeout values, the code in {@code StudentDLL._ValidateTesteeAccessProc_SP} on line 1117
+     *     specifically queries the {@code session.timelimits} view for the client name where the assessment id is null.
+     *     The assumption here is that the TA Check-In time is always configured at the client level, where other time
+     *     limits can be configured at the assessment level.  Furthermore, the TA Check-In time setting at the client
+     *     level supercedes any setting at the test level.  This is the only value in {@code session.timelimits} that
+     *     exhibits this behavior.
+     * </p>
+     * <p>
+     *     Because of the logic in the legacy Student code, the TA Check-In time limit can never be null.  If a check in
+     *     value is not found in the database, it is set to 0.
+     * </p>
+     * <p>
+     *     Because these values will rarely change, it is a strong candidate for caching.
+     * </p>
+     *
+     * @param clientName The name of the client for which the TA Check-In Time should be fetched.
+     * @return An {@code int} representing the length of the TA Check-In timeout window in minutes.  If no value is
+     *         present for the specified client name, 0 will be returned.
+     */
+    Optional<Integer> getTaCheckInTimeLimit(String clientName);
```

Comment:
If the TA Check-In time limit can never be null then why do we return an optional?  Shouldn't we just return 0 and allow this to be a primitive?